File: backend/leon-service/src/main/java/pl/leon/form/application/leon/model/response/forms/FormUiUxTypeRandomTimeStatisticsResponse.java
package pl.leon.form.application.leon.model.response.forms;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import pl.leon.form.application.leon.core.enums.FormLevelType;
import pl.leon.form.application.leon.model.response.questions.QuestionUiUxStatisticsResponse;

import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class FormUiUxTypeRandomTimeStatisticsResponse {
    private FormLevelType uiLevel;
    private FormLevelType uxLevel;
    private List<QuestionUiUxStatisticsResponse> averageTimeToAnswerPerQuestionType;
    private long answerCount;
}

File: frontend/src/app/form-random-statistics/models/form-ui-ux-type-random-time-statistics-response.ts
import { QuestionUiUxStatisticsResponse } from "./question-ui-ux-statistics-response";

export class FormUiUxTypeRandomTimeStatisticsResponse {
    uiLevel: string;
    uxLevel: string;
    averageTimeToAnswerPerQuestionType: QuestionUiUxStatisticsResponse[];
    averageTimeToFillForm: number;
    answerCount: number;
}